### PR TITLE
Reserve capacity for the answer of common_prefix_search

### PR DESCRIPTION
**Pull request by MnO2**
* Justification of this change is here: https://github.com/messense/jieba-rs/issues/19#issuecomment-504662197

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -130,6 +130,9 @@ impl<'a> DoubleArrayTrieBuilder<'a> {
 
     // pub fn build(mut self, keys: &[&str], values: &[usize]) -> DoubleArrayTrie {
     pub fn build(mut self, keys: &'a [&str]) -> DoubleArrayTrie {
+        //using the unicode scalar len is correct since that's our DARTS unit here
+        let longest_word_len = keys.iter().map(|s| s.len()).max().unwrap_or(0);
```
Comment (andelf) [inline src/lib.rs]: ``common_prefix_search`` returns ``char`` code points. ``.chars().count()`` may be a better choice.

File: src/lib.rs [hunk 2]
```diff
@@ -334,7 +339,11 @@ impl DoubleArrayTrie {
 
     /// Find all matched prefixes. Returns [(end_index, value)].
     pub fn common_prefix_search(&self, key: &str) -> Option<Vec<(usize, usize)>> {
-        let mut result = vec![];
+        //reserve the capacity to speed up the performance,
```
Comment (andelf) [inline src/lib.rs]: leading space in comments

**Comment by andelf** [comment]
Good point! Since ``Vec::push`` is sometimes expensive (growing memory and moving items).
``str::len()`` is too large. Using char-size might be a good choice.
Median or mode may be acceptable as well.

File: src/lib.rs (hunk 2 above)
Comment (andelf) [reply src/lib.rs]: cargo-fmt it! :)

**Comment by MnO2** [comment]
Updated. :-)

**Pull request merged**